### ISSUE TITLE
Option to disable the announcement "Copied to clipboard"

### ISSUE THREAD

**Issue by DrSooom** [nvaccess/nvda#9273]
### Path:
NVDA menu » Preferences » Settings... » Browse Mode
### Steps to reproduce:
1. Select a file or folder in the Windows Explorer, press F2 followed by CTRL+C.
2. Select this text in a web browser like Firefox and press CTRL+C.
### Actual behaviour:
1. Nothing is announced.
2. "Copied to clipboard" is announced via speech and braille (if braille message timeout isn't set to "0").
#### Solution No. 1:
Add a checkbox called "Report when a selected text has copied to the clipboard in browse mode" in the Browse Mode NVDA Settings. This checkbox is enabled by default (= current behaviour).
#### Solution No. 2:
Add a combobox called "Report when a selected text has copied to the clipboard" to the Browse Mode NVDA Settings with the following values.
+ Never
+ Only in browse mode (virtual documents/web pages) [default value, current behaviour]
+ Only in focus mode (applications)
+ Always
#### Reasons:
1. Inconsistent behaviour which could confuse casual users and beginners.
2. This "useless" announcement could ride on advanced/power users' nerves, because it always interrupts their workflow.
#### NVDA installed/portable:
Both
#### NVDA version:
2018.4.1
#### Windows version:
Win7 to Win10-1903
#### Name and version of other software in use when reproducing the issue:
All supported Browsers, Adobe Reader, Foxit Reader etc.

**Comment by Brian1Gaff** [comment]
I thought nobody had perfected this to work reliably anyway, which is why I 
use an add on to tell me. It does not mean its copied of course it means 
that you pressed the right keys.
  The only time it natively says this is using the screen and  global nav 
keys to copy to  the buffer.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by lukaszgo1** [comment]
I definitely agree with @DrSooom  we should have a possibility do  disable this useless announcementn in browse mode.

**Comment by netblue44** [comment]
Lets make shore that this very useful feature is not eliminated altogether

in fact announcing coppied and pasted itomms should be included in NVDA core

**Comment by leonardder** [comment]
As for item 1, this is a duplicate of #553. Cutting and copying actions could not reliably be tracked.

As for disabling the copy feedback in browse mode, this is not very likely to receive high priority. Giving this a p4

**Comment by fernando-jose-silva** [comment]
I am in favor of giving the user the choice if they want to hear copied and pasted or even trimmed.
I'm giving classes now, and users always get confused if they press the right keys because nvda is muted when copying a file, folder, etc.
My students for other times have problems other than vision, such as motor coordination, and intellectual disability.

**Comment by lukaszgo1** [comment]
Do you think that NV Access would accept PR adding a possibility to disable this announcement? I am under the impression that for main developer less options the better, but if such feature would be approved I would probably submit a PR for this.

**Comment by larry801** [comment]
@fernando-jose-silva   You can try clipSepak addon if you want feedback when copying file.

**Comment by leonardder** [comment]
> 
> 
> Do you think that NV Access would accept PR adding a possibility to disable this announcement? I am under the impression that for main developer less options the better, but if such feature would be approved I would probably submit a PR for this.

I could not speak on behalf of NV Access, but I guess such a pr makes sense, as this issue points out that a majority seems to agree that this functionality would be useful.

**Comment by JulienCochuyt** [comment]
If #9843 is accepted, implementing configurability of control+C announcement in browse mode will be a one-liner (+ GUI).
In `CursorManager.script_copyToClipboard`, line 412, change
`info.copyToClipboard(notify=True)`
to something like
`info.copyToClipboard(notify=config.conf["virtualBuffers"]["notifyOnCopy"])`

While I am in favor of a clear message being presented for other copy operations like NVDA+F10*2 or NVDA+T*3, I feel that presenting a message on control+C in browse mode makes beginners feel uncomfortable when no message is presented in other apps, as we cannot reliably present it.
It even went as far as a popular addon circumventing this situation by echoing "Copied to clipboard" in every situations, including when there was no selection to copy or control+C was used to cancel an running operation in a console - very bad fix IMHO.

**Comment by Andre9642** [comment]
@JulienCochuyt Do you plan to implement the possibility to disable the announcement soon? The merge of #9843 is a big regression for me and I've already patched the core to prevent the announcement of the text copied to clipboard in browse mode. Indeed when the selection contains a large text and I press CTRL+c, NVDA lags completely. Sometimes pressing Control key doesn't stop speech. Or the synthesizer no longer work during a few seconds, it's very annoying.
In general, I don't understand the interest of the announcement of the text copied. We already have a command to reads any currently selected text (default: NVDA+Shift+upArrow or NVDA+shift+s). This is sufficient IMHO.

**Comment by feerrenrut** [comment]
Yes, we will accept a PR to allow disabling this announcement, however the difficulty will be working out where the option should be located and the wording of it. A new panel or section may need to be created. Rather than jump to a specific setting for 'copied to clipboard', I'd prefer a generic option for "Speech UI verbosity" with options "low", "normal". In future we could add an option for "custom" that shows a checkable list with all the different messages.

**Comment by josephsl** [comment]
Hi, I think a new “verbosity” panel would be a good idea to gather together verbosity level and settings like the one being proposed. Thanks.

**Comment by feerrenrut** [comment]
Or perhaps 'speech presentation': settings related to NVDA's speech UI

Longer term, we hope to also look at customizing the order (perhaps contents) of these UI messages. If we did this, we'd need a mirror panel for 'braille presentation': settings related to NVDA braill UI

**Comment by josephsl** [comment]
Hi, one possibility is adding a button inside current “speech” and “braille” panels that will open speech/braille UI messages dialog, similar to speech synthesizer and braille display selection mechanism. Thanks.

**Comment by feerrenrut** [comment]
Maybe, but I think it would be good to separate settings for devices (synth, braille, audio) from settings for UI (speech UI, braille UI)

**Comment by CyrilleB79** [comment]
@Andre9642 wrote:
> @JulienCochuyt Do you plan to implement the possibility to disable the announcement soon? The merge of #9843 is a big regression for me and I've already patched the core to prevent the announcement of the text copied to clipboard in browse mode. Indeed when the selection contains a large text and I press CTRL+c, NVDA lags completely. Sometimes pressing Control key doesn't stop speech. Or the synthesizer no longer work during a few seconds, it's very annoying.

I have opened #11843 for this regression. Feel free to comment here.
Let's keep this issue to talk about the verbosity level.

**Comment by CyrilleB79** [comment]
I think that UIA notifications in recent MS Office versions just report "Copied", to be checked.

Shortening our "Copied to clipboard" to "Copied" would already make the message shorter for people complaining of this too long message and would make something consistent with other "Copied" messages, even if they are not ours.

What do you think?

**Comment by Adriani90** [comment]
I agree the message "copied" should suffice for all instances where ctrl+c command is used, also in browse mode.
